web-console/src/dialogs/doctor-dialog/doctor-checks.tsx

```diff
@@ -0,0 +1,421 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import axios from 'axios';
+
+import { pluralIfNeeded, queryDruidSql } from '../../utils';
+import { deepGet } from '../../utils/object-change';
+import { postToSampler } from '../../utils/sampler';
+
+export interface CheckControls {
+  addSuggestion: (message: string) => void;
+  addIssue: (message: string) => void;
+  terminateChecks: () => void;
+}
+
+export interface DoctorCheck {
+  name: string;
+  check: (controls: CheckControls) => Promise<void>;
+}
+
+const RUNTIME_PROPERTIES_ALL_NODES_MUST_AGREE_ON: string[] = [
+  'user.timezone',
+  'druid.zk.service.host',
+];
+
+const RUNTIME_PROPERTIES_ALL_NODES_SHOULD_AGREE_ON: string[] = ['java.version'];
+
+// In the future (when we can query other nodes) is will also be cool to check:
+// 'druid.storage.type' <=> historicals, overlords, mm
+// 'druid.indexer.logs.type' <=> overlord, mm, + peons
+
+const RUNTIME_PROPERTIES_MASTER_NODES_SHOULD_AGREE_ON: string[] = [
+  'druid.metadata.storage.type', // overlord + coordinator
+  'druid.metadata.storage.connector.connectURI',
+];
+
+export const DOCTOR_CHECKS: DoctorCheck[] = [
+  // -------------------------------------
+  // Self (router) checks
+  // -------------------------------------
+  {
+    name: 'Verify own status',
+    check: async controls => {
+      // Make sure that the router responds to /status and gives some valid info back
+      let status: any;
+      try {
+        status = (await axios.get(`/status`)).data;
+      } catch (e) {
+        controls.addIssue(
+          `Did not get a /status response, is the cluster running? Got: ${e.message}`,
+        );
+        controls.terminateChecks();
+        return;
+      }
+
+      if (typeof status.version !== 'string') {
+        controls.addIssue('Could not get a valid /status response.');
+      }
+    },
+  },
+  {
+    name: 'Verify own runtime properties',
+    check: async controls => {
+      // Make sure that everything in /status/properties is above board
+      let properties: Record<string, string>;
+      try {
+        properties = (await axios.get(`/status/properties`)).data;
+      } catch (e) {
+        controls.addIssue('Did not get a /status/properties response, something must be broken.');
+        return;
+      }
+
+      // Check that the management proxy is on, it really should be for someone to access the console in the first place but everything could happen
+      if (properties['druid.router.managementProxy.enabled'] !== 'true') {
+        controls.addIssue(
+          `The router's "druid.router.managementProxy.enabled" is not reported as "true" that is unusual.`,
```

Comment:
I would go with different language, like telling the user that we recommend setting this property in order for Coordinator / Overlord APIs to be accessible through the Router (and therefore the web console).

The general idea is we should give people some hints about why each check is there and what they should do about them if issues are found.